### PR TITLE
LXD returns untyped errors

### PR DESCRIPTION
**Pull request by SimonRichardson**
## Description of change

LXD can return untyped errors, so we can't identify when something
is not found or has been successfully deleted.

## QA steps

```
juju bootstrap lxd test
juju deploy redis
juju kill-controller -t 0s -y test
```

### CODE REVIEW

**Comment by SimonRichardson** [comment]
https://trello.com/c/pJTCBbJN/1206-lxd-returns-not-found-error-when-it-cant-find-container

File: container/lxd/container.go [hunk 1]
```diff
@@ -379,3 +383,10 @@ func containerHasStatus(container api.Container, statuses []string) bool {
 	}
 	return false
 }
+
+// lxdNotFound is here because LXD doesn't use typed errors, which causes some
+// pain the in the juju code base. We essentially need to do a string match
+// based on https://github.com/lxc/lxd/blob/097d4845ea5ee78cee5fde6451466d7fd9225726/lxd/response.go#L474
+func lxdNotFoundErr(err error) bool {
```
Comment (manadart) [inline container/lxd/container.go]: Let's use `IsLXDNotFound` in server.go

Comment (SimonRichardson) [reply container/lxd/container.go]: Ah, didn't know that existed, :star:

**Comment by SimonRichardson** [comment]
$$merge$$

**Comment by SimonRichardson** [comment]
$$merge$$

> workerSuite.TestRemovePrimaryValidSecondaries

**Pull request merged**